tools/codegen/codespec/testdata/api-spec.yml

```diff
@@ -376,20 +404,30 @@ components:
           items:
             type: string
       required:
-        - innerStrAttr
         - innerNumAttr
     NestedTestResourceRequest:
       type: object
       properties:
-        nestedObjectArrayAttr:
+        nestedListArrayAttr:
           type: array
           description: Test field description
           items:
             $ref: "#/components/schemas/NestedObjectAttr"
           maxItems: 1
           minItems: 1
       required:
-        - nestedObjectArrayAttr
+        - nestedListArrayAttr
+    SimpleStringRefObject:
+      type: string
+      description: Test field description
     NoBody:
       type: object
       description: Endpoint does not return a response body.
+    NestedMapObjectAttr:
+      type: object
+      additionalProperties:
```

Comment:
q: curious if in our api spec there is use of `additionalProperties` for representing generic maps.

Reply:
There is. Connection strings is one example https://github.com/mongodb/atlas-sdk-go/blob/main/openapi/atlas-api-transformed.yaml#L22082

Reply:
great 👍